crates/dojo-world/src/utils.rs

```diff
@@ -328,6 +332,55 @@ pub fn block_number_from_receipt(receipt: &TransactionReceipt) -> u64 {
     }
 }
 
+#[allow(async_fn_in_trait)]
+pub trait TransactionExt<T>
+where
+    T: ConnectedAccount + Sync,
+{
+    type R;
+
+    async fn send_with_cfg(
+        self,
+        txn_config: &TxnConfig,
+    ) -> Result<Self::R, AccountError<T::SignError>>;
+}
+
+impl<T> TransactionExt<T> for Execution<'_, T>
+where
+    T: ConnectedAccount + Sync,
+{
+    type R = InvokeTransactionResult;
+
+    async fn send_with_cfg(
+        mut self,
+        txn_config: &TxnConfig,
+    ) -> Result<Self::R, AccountError<T::SignError>> {
+        if let TxnConfig { fee_estimate_multiplier: Some(fee_est_mul), .. } = txn_config {
+            self = self.fee_estimate_multiplier(*fee_est_mul);
```

Comment:
If we know that `fee_estimate_multiplier` is exclusive to `max_fee_raw`, we should set it to `None` explicitly. To be checked in starknet-rs to ensure we don't have any non-desired behavior here.

Reply:
i think setting it explicitly to `None` won't be a good idea here because it would lead to neither of them getting applied, what we should do instead check that only one of them is `Some` and panic if thats not the case.

either way we don't have `max_fee_raw` in this PR yet, so i have noted to check this in the PR adding `max_fee_raw`